conu/backend/docker/image.py

```diff
@@ -221,6 +226,99 @@ def push(self, repository=None, tag=None):
                                         self.name, error)
         return image
 
+    def set_transport(self, transport=None, path=None):
+        """
+        :param transport: from where will be this image copied
+        :param path in filesystem
+        :return: self
+        """
+        if not transport:
+            return self
+        if self.transport == transport and self.path == path:
+            return self
+        path_required = [Transport.DIRECTORY, Transport.DOCKER_ARCHIVE, Transport.OCI]
+        if transport in path_required:
+            self.path = self.mount(path).mount_point
+        elif transport == Transport.OSTREE:
+            if path and not os.path.isabs(path):
+                raise ConuException("Path '", path, "' for OSTree transport is not absolute")
+            self.path = path
+        else:
+            self.path = None
+
+        self.transport = transport
+        return self
+
+    def save_to(self, image):
+        """ Save this image to another DockerImage
+
+        :param image: DockerImage
+        :return:
+        """
+        if not isinstance(image, self.__class__):
+            raise ConuException("Invalid target image type", type(image))
+        self.copy(image.name, image.tag, target_transport=image.transport, target_path=image.path)
+
+    def load_from(self, image):
+        """ Load from another DockerImage to this one
+
+        :param image:
+        :return:
+        """
+        if not isinstance(image, self.__class__):
+            raise ConuException("Invalid source image type", type(image))
+        image.save_to(self)
+
+    def skopeo_pull(self):
+        """ Pull image from Docker to local Docker daemon using skopeo
+
+        :return: pulled image
+        """
+        return self.copy(self.name, self.tag, Transport.DOCKER, Transport.DOCKER_DAEMON)\
+            .set_transport(Transport.DOCKER_DAEMON)
+
+    def skopeo_push(self, repository=None, tag=None):
+        """ Push image from Docker daemon to Docker using skopeo
+
+        :param repository: repository to be pushed to
+        :param tag: tag
+        :return: pushed image
+        """
+        return self.copy(repository, tag, Transport.DOCKER_DAEMON, Transport.DOCKER)\
+            .set_transport(Transport.DOCKER)
+
+    def copy(self, repository=None, tag=None,
+             source_transport=None,
```

Comment:
souce_transport should match the current image right? so likely it should be `docker-daemon:`

Reply:
I was thinking, that if DockerImagePullPolicy is set to NEVER, it maid not be docker-daemon:, so I let it to be deduced by object transport attribute.
Right now, if source_transport is not None, it will change Transport of copied object.
I am realising that it probably isn’t expected behaviour in copy function, to change copied object. On the other hand it makes sense, because it actually defines where the image is. (image Transport now has set default value in DockerImage constructor, depending on DockerImagePullPolicy, [but it doesn’t change if original pull method is called rn]).

Reply:
right, that makes sense

My intention here was to have a good user experience: filling in a ton of input parameters can feel overwhelming.